variables.tf

```diff
@@ -49,6 +55,18 @@ variable "tags" {
   type        = map(string)
 }
 
+variable "use_keypair" {
```

Comment:
IMO this name doesn't reflect actual behavior. To me this naming is whether a keypair should be used at all, but if you specify an external key it will be used even if this is false. I think this should instead be `create_keypair`